staging/src/k8s.io/client-go/discovery/cached/disk/cached_discovery.go

```diff
@@ -75,7 +75,7 @@ func (d *CachedDiscoveryClient) ServerResourcesForGroupVersion(groupVersion stri
 
 	liveResources, err := d.delegate.ServerResourcesForGroupVersion(groupVersion)
 	if err != nil {
-		klog.V(3).Infof("skipped caching discovery info due to %v", err)
+		klog.V(3).Infof("skipped caching discovery info due to %v", cacheErr)
```

Comment:
Why is this correct? It looks to me that it failed because err wasn't nil?

Reply:
Oh I see

Reply:
I still don't get it. Why should we use the `cacheErr` here? I thought what this log line means is that we are skipping `d.writeCachedFile` at line 86 because the delegate response has err.

Reply:
Also see the log at line 87: `klog.V(1).Infof("failed to write cache to %v due to %v", filename, err)`

Reply:
Really thanks for your apply! But i still think we have to change this:Here is a example

Reply:
Really thanks for your apply! But i still think we have to change this, maybe we can discuss this. Here is a example:
The cache file in `.kube/cache` has 10 minutes  expire time. Now suppose the time has already gone for more than 10 minutes. So when using `cachedBytes, err := d.getCachedFile(filename)` in line 109. It will report the error:
```
	if time.Now().After(fileInfo.ModTime().Add(d.ttl)) {
		return nil, errors.New("cache expired")
	}
```
So this error is not nil in line 111:
```
if err == nil 
```
and this error is reason why we skipped the cache discovery(am i right with this? i'am not sure). But we doesn't report this error because in line 119:
```
liveGroups, err := d.delegate.ServerGroups()
``` 
So we report this as: 
```
klog.V(3).Infof("skipped caching discovery info due to %v", err)
```
And finally no place will report this `cache expired` error.

Reply:
adding another info level log (at a verbosity of 6-8?) on `cacheErr != nil` will solve the case is it?

Reply:
@yue9944882  Thanks. I made few changes for your suggestion.